apps/tailwind-components/components/editor/CodeEditor.vue

```diff
@@ -11,45 +11,33 @@ const props = withDefaults(
   }
 );
 
-const isExpanded = ref<boolean>(true);
-const code = ref<string>(
-  "<div><h1>hello, world!</h1><p>test</p><p>test 1</p></div>"
-);
-const editor = useTemplateRef<HTMLDivElement>("editor");
+const emits = defineEmits<{
+  (e: "update:modelValue", value: string): void;
+}>();
 
-const options = {
-  automaticLayout: true,
-  value: code.value,
-  language: props.lang,
-  theme: "vs-dark",
-  formatOnPaste: true,
-  formatOnType: true,
-  autoIndent: "brackets",
-  autoClosingBrackets: "always",
-  wordWrap: "on",
-  dimension: {
-    height: 310,
-    width: 100,
-  },
-  suggest: {
-    insertMode: "insert",
-  },
-};
+const code = defineModel<string>("");
+const isExpanded = ref<boolean>(true);
+const editor = useTemplateRef<InstanceType<typeof MonacoEditor>>("editor");
 
-onMounted(() => {
-  // monacoEditor.value = monaco.editor.create(editor.value as HTMLElement, {
-  // });
+watchEffect(() => {
+  if (editor.value?.$editor) {
+    const action = editor.value.$editor?.getAction(
+      "editor.action.formatDocument"
+    );
+    if (action) {
+      action.run();
+    }
+  }
 });
 
-// watch(() => monacoEditor.value, () => {
-//   monacoEditor.value.getModel().onDidChangeContent(() => {
-//     code.value = toRaw(monacoEditor.value).getValue()
-//   })
-// });
+watchEffect(() => {
```

Comment:
I suppose the original input is only loaded once, so is this watch really needed?